### ISSUE TITLE
Stomps all over the local keymap if outline-minor-mode is enabled

### ISSUE THREAD

**Issue by phil-s** [hexmode/mediawiki-el#36]
`(add-hook 'outline-minor-mode-hook 'mediawiki-outline-magic-keys)` is really bad.

`mediawiki-outline-magic-keys` calls `local-set-key` for a bunch of key sequences.  That's usually the major mode keymap, which is shared amongst all buffers in that mode whether or not `outline-minor-mode` (let alone `mediawiki-mode`) is enabled in any given buffer, and which persists for future buffer using that major mode as well.

And because the *global* `outline-minor-mode-hook` was used, this clobbering happens again in future, even if you'd re-bound the keys manually, until you figure out what was causing it.

And this all happens from simply *loading* mediawiki.el.

I suggest creating a new buffer-local minor mode for the desired key bindings, and then only enabling that mode in the specific buffers where you want it.

**Comment by hexmode** [comment]
Excellent points.  I'll look at fixing this soon.

**Comment by hexmode** [comment]
Could you see if this is fixed in the latest master?

**Comment by phil-s** [comment]
No, the current master is commit 05aab5ce3b4708dd12e6e33dfab390a9c19dcabf and we can see that nothing has changed:
https://github.com/hexmode/mediawiki-el/blob/05aab5ce3b4708dd12e6e33dfab390a9c19dcabf/mediawiki.el#L1917-L1932

**Comment by hexmode** [comment]
ah, thanks for showing me.

**Comment by aspiers** [comment]
Ouch, this is really bad.  It just bit me by breaking `outshine-mode`, and confused the hell out of me until I traced it to this package.  I suggest simply removing that `add-hook` line as a quick fix, but yeah, these keys absolutely must be moved to their own map.  In the mean time I'm going to have to uninstall this package unfortunately.

**Comment by logological** [comment]
Even with the latest changes, I don't think it's a good idea for this package to override the standard `left-word`, `right-word`, `backward-paragraph`, and `forward-paragraph` bindings.  Many (if not most) users *very frequently* use `control left`, `control right`, `control up`, and `control down` to navigate text. Adjusting MediaWiki outline levels is an action whose comparative rarity doesn't justify overriding a ubiquitous and expected standard.  Aren't there some other keybindings that could be used for adjusting the outline levels instead?

**Comment by hexmode** [comment]
I agree that overriding standard bindings is not a good idea.  That certainly isn't my intention.

FWIW, I use `M-f` and `M-b` for `forward-word` and `backward-word` instead of the the `left-` and `right-` bindings (as well as `C-f` and `C-b`).  Since I avoid the arrow keys, I am not a user of the `-paragraph` bindings.

I'll take another look and see what needs to be done for these keys.
